Source/core/CyclicBuffer.cpp

```diff
@@ -178,7 +218,11 @@ namespace Core {
     void CyclicBuffer::AdminLock()
     {
 #ifdef __POSIX__
+#ifndef __DEBUG__
         pthread_mutex_lock(&(_administration->_mutex));
+#else
+        ASSERT(!pthread_mutex_lock(&(_administration->_mutex)));
```

Comment:
Lets not execute essential code within ASSERT statements. Just pick up the return value (always) and ASSERT on the result.. There are macros, to avoid WARNINGS on none used variables (if the ASSERT implementation is empty) 
`int result = pthread_mutex_lock(&(_administration->_mutex));
ASSERT(result == 0);
DEBUG_VARIABLE(result);`
should do the trick..